samples/net/openthread/coap/README.rst

```diff
@@ -0,0 +1,189 @@
+   router
+   Done
+
+A valid state could be child, router or leader.
+
+Once Thread network is operationnal, you could start using client.
+
+Controlling server board's LED using a button on client board
+*************************************************************
+
+There are nothing to do once Thread network is operationnal.
+Just press the button sw0 on the client and you should see led0 toggling.
+
+The client use a broadcast address to request CoAP server to toggle the LED.
+It does not know the address of the server so if there are a second server
+on the network, then the LED of the second board will toggle.
+
+Controlling server board's LED from a computer
+**********************************************
+
+Although we use OpenThread CoAP API, we could interact with any CoAP client
+or server available on network. In this example, we are going to control the
+LED from a computer that is not in the Thread network.
+This requires an `OpenThread Border Router`_ with NAT64 support enabled on the same network.
+
+First, check that the server (or the client) could is connected to the otbr and
+could use NAT64:
+
+.. code-block::
+
+   server:~$ ot netdata show
+   router
+   Done
+   Prefixes:
+   fd6f:cb3a:802:1::/64 paos low dc00
+   Routes:
+   fc00::/7 sa med dc00
+   fd6f:cb3a:802:2:0:0::/96 sn low dc00
+   Services:
+   44970 01 14000500000e10 s dc00 0
+   44970 5d fd78b9ce54779c6eb5484d062c3b5b22d120 s dc00 1
+   Contexts:
+   fd6f:cb3a:802:1::/64 1 c
+   Commissioning:
+   11426 - - -
+   Done
+
+Prefixes show the IPv6 prefies that could be used by device outside the
+Thread network to contact devices on Thread network.
+
+We should have an IPv6 using the prefixes:
+
+.. code-block::
+
+   server:~$ ot ipaddr
+   fd78:b9ce:5477:9c6e:0:ff:fe00:a800
+   fd6f:cb3a:802:1:f0ec:c1e2:c1bb:744
+   fd78:b9ce:5477:9c6e:75b8:386c:1f79:1013
+   fe80:0:0:0:50d1:bed5:6e6e:ad75
+   Done
+
+fd6f:cb3a:802:1:f0ec:c1e2:c1bb:744 is the IPv6 address that could be used
+to contact the CoAP server outside of the Thread network.
+
+We could also check that we could access internet from Thread network:
+
+.. code-block::
+
+   server:~$ ot ping 8.8.8.8
+   Pinging synthesized IPv6 address: fd6f:cb3a:802:2:0:0:808:808
+   16 bytes from fd6f:cb3a:802:2:0:0:808:808: icmp_seq=1 hlim=114 time=36ms
+   1 packets transmitted, 1 packets received. Packet loss = 0.0%. Round-trip min/avg/max = 36/36.0/36 ms.
+   Done
+
+If everything is working, then, we could start controlling the LED from a computer.
+To do that, let's use aiocoap-client, a tool written in python.
+First, install it:
+
+.. code-block::
+
+   pip install aiocoap
+
+
+Then, send a request to the server to toggle the LED:
+
+.. code-block::
+
+    aiocoap-client -m PUT --payload '{"led_id":0,"state":2}'  coap://[fd6f:cb3a:802:1:f0ec:c1e2:c1bb:744]/led
+
+The LED state should have changed.
+
+
+.. _OpenThread Border Router: https://openthread.io/codelabs/openthread-border-router-nat64?hl=fr#1
```

Comment:
The link auto-translates the page to French, I don't think it's intended. I guess we'd need to drop `?hl=fr#1` param.